Improve layout on Add/Edit Author page

sandriu opened the pull request:
Fields grouped by role, better spacing. Some bugs still present for Relations table.

CatalinFrancu commented:
Looks beautiful! A few thoughts:

* Do you think it's worth using the `<fieldset>` element instead of the `.common-region` class? It is logically clearer. Alternatively, we could rename `common-region` to something like `field-set` to clarify that it is intended for grouping fields.
* Along the same lines, I would rename `.special-column` to something that makes its purpose clearer.
* I will check to see if the Select2 styles you want can be achieved via its options (with less CSS).

sandriu commented:
I replaced the `<div>` elements called 'common-region' with `<fieldset>` elements, for a clearer code. However, the fieldsets still require a custom CSS class in order to display a visual way of grouping similar fields in the interface (a vertical gray line, in our case), because Bootstrap 4 `<fieldset>` does not come with any border or padding or margin https://getbootstrap.com/docs/4.4/content/reboot/#forms.

The need for grouping similar fields in a visual way is based on **Gestalt's Proximity Principle** https://www.usertesting.com/blog/gestalt-principles#proximity, which states that '_things that are close together appear to be more related than things that are spaced farther apart_'.
So I decided to use the similarity between some fields in order to divide the long page "Add Author" (and "Edit Author") into smaller chunks that are easier to complete and follow.

![screenshot_Edit_Author_aerial_view](https://user-images.githubusercontent.com/1265882/80477681-0eb21a00-8955-11ea-8625-824eabb556ae.png)

For a clearer meaning, I also renamed 'common-region' class as 'related-fields' and 'special-column' class as ''narrow-column'.
